src/GafferSceneUI/LightTool.cpp

```diff
@@ -760,495 +792,438 @@ class LightToolHandle : public Handle
+			Inspector::ResultPtr inspection = nullptr;
+			float originalValue = 0;
+
+			auto it = m_inspectors.find( metaParameter );
+			if( it == m_inspectors.end() )
 			{
 				return false;
 			}
 
-			// We can be called to check visibility for any scene location set in the current context, spot light
-			// or otherwise. If there isn't an inspection, this handle should be hidden (likely because the scene
-			// location is not a spot light).
+			if( inspection = it->second->inspect() )
+			{
+				auto originalData = runTimeCast<const FloatData>( inspection->value() );
+				assert( originalData );
+				originalValue = originalData->readable();
+				info = { inspection, originalValue };
+				return true;
+			}
 
-			Inspector::ResultPtr contextConeInspection = m_coneAngleInspector->inspect();
-			Inspector::ResultPtr contextPenumbraInspection = m_penumbraAngleInspector ? m_penumbraAngleInspector->inspect() : nullptr;
+			return false;
+		}
 
-			if( !contextConeInspection || ( m_handleType == HandleType::Penumbra && !contextPenumbraInspection ) )
+		bool allInspectionsEnabled() const
+		{
+			std::vector<const Inspector::Result *> requiredInspections = handleValueInspections();
+			if( requiredInspections.empty() )
 			{
 				return false;
 			}
 
-			// We are a spot light, but the penumbra will be hidden if it's too close to the cone angle, for
-			// the location we're attaching the handles to.
+			bool enabled = true;
+			for( const auto &i : requiredInspections )
+			{
+				enabled &= i && i->editable();
+			}
 
-			/// \todo This checks the penumbra / cone angles only for the last selected location, causing
-			/// repeated checks of the same location when `visible()` is called in a loop over multiple scene
-			/// locations. We rely on history caching to make this relatively fast, but ideally this could be
-			/// tested only once per selection list.
+			return enabled;
+		}
 
-			const auto &[coneInspection, coneAngle, penumbraInspection, penumbraAngle] = spotLightHandleAngles();
-			if( m_handleType == HandleType::Penumbra && penumbraAngle )
-			{
-				const float radius = m_visualiserScale * m_frustumScale * -10.f;
-				const V2f coneRaster = m_view->viewportGadget()->gadgetToRasterSpace(
-					V3f( 0, 0, radius ),
-					this
-				);
-				const M44f rot = M44f().rotate( V3f( 0, degreesToRadians( penumbraAngle.value() ), 0 ) );
-				const V2f penumbraRaster = m_view->viewportGadget()->gadgetToRasterSpace(
-					V3f( 0, 0, radius ) * rot,
-					this
-				);
+		// May be overriden to update internal state based on the `scenePath`
+		virtual void setScenePathInternal( ScenePathPtr scenePath )
+		{
 
-				if( ( coneRaster - penumbraRaster ).length() < ( 2.f * g_circleHandleWidthLarge ) )
-				{
-					return false;
-				}
-			}
+		}
 
-			return true;
+		// May be overriden to clean up internal state after a drag.
+		virtual bool handleDragEndInternal()
+		{
+			return false;
```

Comment:
A whole load of these virtual functions intended to be overridden are in the `private` section, but I'd expect them to be in the `protected` section.

Reply:
By way of documenting our offline conversation, I was originally making these private in the vein discussed here : http://www.gotw.ca/publications/mill18.htm where implementation, even for virtual methods, are encouraged to be kept `private` unless they need to be called from a derived class, in which case they would be `protected`.

That doesn't fit with the pattern in other classes though, and could also cause problems should we ever want to bind them in Python. So I've made them `protected` in https://github.com/GafferHQ/gaffer/pull/5483/commits/98d314c1bbc7aaa560a05047f1f80757f312d7be